src/definition/accessors/IConfigurationExtend.ts

```diff
@@ -17,4 +18,6 @@ export interface IConfigurationExtend {
     /** Accessor for declaring the commands which your App provides. */
     readonly slashCommands: ISlashCommandsExtend;
     readonly api: IApiExtend;
+
+    readonly externalComponents: IExternalComponentsExtend;
```

Comment:
Another place where the terminology is consistent with the codebase yet the logic is not consistent. Plural is staying consistent yet the logic of being singular is not consistent since our other logic, for slash commands, is also plural.

Reply:
Sorry, according to the previous discussions (one app only owns one external component). I think here it should be `externalComponent: IExternalComponentExtend`.

Reply:
@lolimay is there a technical reason why we wouldn't allow an app to register more than one `externalComponent`? For me, considering the functionality, an app could register a MODAL component to show at some point in time and a CONTEXTUAL_BAR in another - it doesn't look like there should be a problem